prow/metrics/http.go

```diff
@@ -88,14 +108,49 @@ func (trw *traceResponseWriter) Write(data []byte) (int, error) {
 	return size, err
 }
 
+// Metrics holds the metrics for Prometheus
+type Metrics struct {
+	HTTPRequestDuration *prometheus.HistogramVec
+	HTTPResponseSize    *prometheus.HistogramVec
+	ErrorRate           *prometheus.CounterVec
+}
+
+// NewMetrics is a constructor for Metrics
+func NewMetrics(namespace string) *Metrics {
+	m := &Metrics{
+		HTTPRequestDuration: HttpRequestDuration(namespace, 0.0001, 2),
+		HTTPResponseSize:    HttpResponseSize(namespace, 256, 65536),
+		ErrorRate:           ErrorRate(namespace),
+	}
+	prometheus.MustRegister(m.HTTPRequestDuration)
+	prometheus.MustRegister(m.HTTPResponseSize)
+	prometheus.MustRegister(m.ErrorRate)
+	return m
+}
+
+// RecordError records the error to prometheus
+func RecordError(label string, errorRate *prometheus.CounterVec) {
+	labels := prometheus.Labels{"error": label}
+	if errorRate != nil {
```

Comment:
Why the nilccheck? Funcs should not assume that they will get nil args, unless its explicitly documented on the func and has good reasons.

Reply:
Ok